x-pack/plugins/observability_solution/synthetics/public/apps/synthetics/components/alerts/common/fields.tsx

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { FieldPopoverExpression } from './field_popover_expression';
+import { Suggestion } from '../hooks/use_fetch_synthetics_suggestions';
+import { FieldSelector } from './field_selector';
+
+interface FieldProps {
```

Comment:
It's likely we could refactor this and achieve the same result with less code. I also note we don't have a ton of module-level testing for this, which is ok for now but would be nice to have down the line, as part of the same refactor.